lib/scala/bench-processor/src/main/java/org/enso/benchmarks/processor/SpecCollector.java

```diff
@@ -282,36 +285,23 @@ private void generateClassForGroup(
             + " read: \" + Objects.toString(projectRootDir));");
     out.println("    }");
     out.println("    File languageHomeOverride = Utils.findLanguageHomeOverride();");
-    out.println("    var ctx = Context.newBuilder()");
-    out.println("      .allowExperimentalOptions(true)");
-    out.println("      .allowIO(IOAccess.ALL)");
-    out.println("      .allowAllAccess(true)");
-    out.println("      .option(RuntimeOptions.LOG_LEVEL, Level.FINE.getName())");
-    out.println("      .logHandler(System.err)");
-    out.println("      .option(");
-    out.println("        RuntimeOptions.LANGUAGE_HOME_OVERRIDE,");
-    out.println("        languageHomeOverride.getAbsolutePath()");
-    out.println("      )");
-    out.println("      .option(");
-    out.println("        RuntimeOptions.PROJECT_ROOT,");
-    out.println("        projectRootDir.getAbsolutePath()");
-    out.println("      )");
+    out.println("    var fallbackLogHandler = JulHandler.get();");
+    out.println("    var logHandler = new LogHandler(fallbackLogHandler);");
     out.println(
         """
-                      .option("engine.TraceCompilation", "true")
-                      .logHandler(new java.util.logging.Handler() {
-                         @Override
-                         public void publish(LogRecord lr) {
-                           if ("engine".equals(lr.getLoggerName())) {
-                             messages.add(lr);
-                           }
-                         }
-                         @Override public void flush() {}
-                         @Override public void close() {}
-                      })
-                """);
-    out.println("      .build();");
-    out.println("    ");
+                     Map<String, String> options = Map.of(
+                         "engine.TraceCompilation", "true",
+                         RuntimeOptions.LANGUAGE_HOME_OVERRIDE, languageHomeOverride.getAbsolutePath()
+                     );
+        """);
+    out.println("    var ctxFactory = ContextFactory.create();");
+    out.println("    ctxFactory.enableStaticAnalysis(false);");
+    out.println("    ctxFactory.enableDebugServer(false);");
+    out.println("    ctxFactory.projectRoot(projectRootDir.getAbsolutePath());");
+    out.println("    ctxFactory.options(options);");
+    out.println("    ctxFactory.logHandler(logHandler);");
+    out.println("    var ctx = ctxFactory.build();");
```

Comment:
[ctxFactory.build()](https://github.com/enso-org/enso/blob/e5b48f4d919a3f1239c7daf25a3e6d0983605248/engine/common/src/main/java/org/enso/common/ContextFactory.java#L218) ensures that Enso logging infrastructure is initialized.

Reply:
As well as ensuring the consistency in how we are building `org.graalvm.polyglot.Context` from CLI, Language server, and from benchmarks.